pandas/tests/indexes/datetimes/test_arithmetic.py

```diff
@@ -307,6 +308,17 @@ def test_dti_cmp_list(self):
 
 class TestDatetimeIndexArithmetic(object):
 
+    # -------------------------------------------------------------
+    # Invalid Operations
+
+    @pytest.mark.parametrize('other', [3.14, np.array([2.0, 3.0])])
```

Comment:
is there a reason to add these to each rather than using a common test?

Reply:
I do slightly prefer the readability when they are in the sub-classes, but at the same time it doesn't fully guarantee that the tests are actually testing all subclasses, so we have a tradeoff here. ideally would systematically test common behaviors in test_datetimelike

Reply:
I thought about this, put them here because ATM this is these are well-established as the correct place for arithmetic tests.  But I agree that there are a bunch of arithmetic tests that could be shared between DTI/TDI/PI.  Maybe tests.indexes.test_datetimelike_arithmetic?  A bit verbose...

BTW next time I complain about parametrization/fixtures, point out statsmodels' tests.

Reply:
ok, let's think about this. maybe open an issue about it.